WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift

```diff
@@ -113,6 +125,44 @@ final class PaymentCaptureOrchestrator: PaymentCaptureOrchestrating {
         stores.dispatch(paymentAction)
     }
 
+    func retryPayment(for order: Order,
+                      onCompletion: @escaping (Result<CardPresentCapturedPaymentData, Error>) -> Void) {
+        guard let handlers = handlersForActivePayment else {
+            return onCompletion(.failure(PaymentCaptureOrchestratorError.couldNotRetryPayment))
+        }
+
+        handlers.onPreparingReader()
+
+        let retryPaymentAction = CardPresentPaymentAction.retryPayment(
+            siteID: order.siteID,
+            orderID: order.orderID,
+            onCardReaderMessage: { event in
+                switch event {
+                case .waitingForInput(let inputMethods):
+                    handlers.onWaitingForInput(inputMethods)
+                case .displayMessage(let message):
+                    handlers.onDisplayMessage(message)
+                case .cardDetailsCollected, .cardRemovedAfterClientSidePaymentCapture:
+                    handlers.onProcessingMessage()
+                default:
+                    break
```

Comment:
Should we log something internally inside the default case here? Just to avoid failing silently if something goes unexpected. 

Or I wonder if we should be more explicit and add the extra cases with an assertion, so if we land in those or we modify the `CardReaderEvent` in the future we'll be warned by the compiler that cases are not being handled:

```
case .cardInserted, .cardRemoved, .lowBattery, .lowBatteryResolved, .disconnected:
   assertionFailure("Unexpected case")
```

Reply:
I don't think they're assertionFailures, really... we continue listening for further events and recover where possible. I've added a similar case handler to `collectPayment` and `retryPayment` which logs at info level, because I agree having the cases listed is better than a default here.